hca/dss/__init__.py

```diff
@@ -1,172 +1,139 @@
-
-    :param uuid: A RFC4122-compliant ID for the bundle.
-    :param bundle_uuid: A RFC4122-compliant ID for the bundle that contains this file.
-    :param creator_uid: User ID who is creating this file.
-    :param source_url: Cloud URL for source data.
-    :param version: Timestamp of file creation in RFC3339.  If this is not provided, the latest version is returned.
-    """
-    return PutFiles.run(locals())
-
-
-def put_subscriptions(callback_url, replica, es_query, **kwargs):
-    """
-    Create a new event subscription. Every time a new bundle version matches this query,
-    a request is sent to callback_url.
-
-    :param callback_url: Url to send request to when a bundle comes in that matches this query.
-    :param es_query: Elasticsearch query that will trigger the callback.
-    :param replica: Replica to write to. Can be one of aws, gcp, or azure.
-    """
-    return PutSubscriptions.run(locals())
-
-
-def download(uuid, replica='aws', name=None, **kwargs):
-    """
-    Download a full bundle or file to local.
-
-    :param uuid: UUID of the folder or file you want to download.
-    :param name: Name of the folder you're storing the bundle/file in (Your name for the bundle). Defaults to the uuid of the bundle or file.
-    :param replica: Which cloud to download from first. One of 'aws', 'gcp', or 'azure'.
-    """
-    return Download.run(locals())
-
-
-def login(access_token=None, **kwargs):
-    """
-    Setup authentication credentials. If you give access_token and refresh_token then these will be saved to config file. If not and you're using a terminal, a browser will pop up for you to sign in to Google to grant oauth credentials. This is the recommended usage pattern. Otherwise, this will raise an exception.
-
-    :param access_token: A Google OAuth access token with https://www.googleapis.com/auth/userinfo.email scope Using this parameter will set the config access_token, but when it has expired there it will not automatically refresh.
-    """
-    return Login.run(locals())
-
-
-def upload(file_or_dir, staging_bucket=None, replica='aws', **kwargs):
-    """
-    Upload a file or directory to the cloud, register each file, and bundle them together.
-
-    :param file_or_dir: List of paths to local/cloud files or a local directory to upload.
-    :param replica: Which cloud to upload to first. One of 'aws', 'gcp', or 'azure'.
-    :param staging_bucket: Bucket within replica to upload to.
-    """
-    return Upload.run(locals())
+from __future__ import absolute_import, division, print_function, unicode_literals
+
+import os
+import time
+import uuid
+from io import open
+
+import requests
+
+from ..util import SwaggerClient
+from ..util.exceptions import SwaggerAPIException
+from .. import logger
+from .upload_to_cloud import upload_to_cloud
+
+class DSSClient(SwaggerClient):
+    """
+    Client for the Data Storage Service API.
+    """
+    UPLOAD_BACKOFF_FACTOR = 1.618
```

Comment:
out of curiosity where does this magic number come from?

Reply:
It doesn't really matter, the exponential backoff factor for HTTP request backoffs can be any number not too far from 2.